transactions/bitcoin/enhancedPsbt.ts

```diff
@@ -0,0 +1,167 @@
+    const output = transaction.getOutput(outputIndex);
+
+    if (!output?.script) {
+      throw new Error('Output with no script detected in PSBT');
+    }
+
+    const outputScript = btc.OutScript.decode(output.script);
+
+    const btcNetwork = this._context.network === 'Mainnet' ? btc.NETWORK : btc.TEST_NETWORK;
+
+    if (outputScript.type === 'unknown') {
+      //for script outputs
+      return {
+        script: btc.Script.decode(outputScript.script).map((i) => (i instanceof Uint8Array ? hex.encode(i) : `${i}`)),
+      };
+    }
+
+    return { address: btc.Address(btcNetwork).encode(outputScript) };
+  }
+
+  async getSummary() {
+    const transaction = btc.Transaction.fromPSBT(this._psbt);
+
+    const inputs: { extendedUtxo: ExtendedUtxo; sigHash?: btc.SigHash }[] = [];
+    const outputs: (TransactionOutput | TransactionScriptOutput)[] = [];
+
+    let isSigHashAll = false;
+    let hasSigHashNone = false;
+
+    let inputTotal = 0;
+    let outputTotal = 0;
+
+    for (let inputIndex = 0; inputIndex < transaction.inputsLength; inputIndex++) {
+      const inputRaw = transaction.getInput(inputIndex);
+      const inputTxid = hex.encode(inputRaw.txid!);
+
+      const input = await this._context.getUtxoFallbackToExternal(`${inputTxid}:${inputRaw.index}`);
+
+      if (!input?.extendedUtxo) {
+        throw new Error(`Could not parse input ${inputIndex}}`);
+      }
+
+      const sigHash = inputRaw.sighashType;
+      inputs.push({
+        extendedUtxo: input.extendedUtxo,
+        sigHash,
+      });
+
+      inputTotal += input.extendedUtxo.utxo.value;
+
+      isSigHashAll = isSigHashAll || sigHash === undefined || sigHash === btc.SigHash.ALL;
+      hasSigHashNone = hasSigHashNone || (sigHash && sigHash & btc.SigHash.NONE) === btc.SigHash.NONE;
+    }
+
+    let currentOffset = 0;
+    for (let outputIndex = 0; outputIndex < transaction.outputsLength; outputIndex++) {
+      const outputMetadata = this.parseAddressFromOutput(transaction, outputIndex);
+
+      if (outputMetadata.script !== undefined) {
+        outputs.push({
+          script: outputMetadata.script,
+        });
+        continue;
+      }
+
+      const outputRaw = transaction.getOutput(outputIndex);
+      const amount = Number(outputRaw.amount);
+      outputTotal += amount;
+      const output: TransactionOutput = {
+        address: outputMetadata.address,
+        amount: Number(amount),
+        inscriptions: [],
+        satributes: [],
+      };
+      outputs.push(output);
+
+      if (!isSigHashAll) {
+        continue;
+      }
+
+      const { inscriptions, satributes } = output;
+
+      let runningOffset = 0;
+      for (const input of inputs) {
+        if (runningOffset + input.extendedUtxo.utxo.value > currentOffset) {
+          const inputBundleData = await input.extendedUtxo.getBundleData();
+          const fromAddress = input.extendedUtxo.address;
+
+          const outputInscriptions = inputBundleData?.sat_ranges
```

Comment:
We have the same code here in the EnhancedTransaction class. We should create a util for it.

Reply:
Ah, thanks for picking this up 🙌 This was a TODO that I never got around to 😅